Implement scoring matrix downloading process

clonder opened the pull request:
Automatic Downloading: Matrix files are automatically downloaded if missing, ensuring up-to-date resources.
Caching: Downloaded files are cached locally, reducing redundant downloads and improving performance.

clonder committed:
Implement scoring matrix downloading process

clonder committed:
Update functions.py

Accio commented:
Thanks @clonder! I will have a look.

alexandra-valeanu commented:
Hey, @clonder! Thank you for working on Kinex!

I tested the changes you made and encountered some errors:
<img width="761" alt="IMG_7947" src="https://github.com/user-attachments/assets/fdab6f49-d2b3-400c-b3d2-eb8a0623e22a">

Consider adding "requests" to  dependencies in `pyproject.toml`

After I install the "requests" package I get the following:
<img width="642" alt="IMG_5138" src="https://github.com/user-attachments/assets/ab3b3e9b-e60c-4e4c-b292-624eaa67de7d">

Here is a link to how the [project.urls] are used: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#urls
Consider adding the links to `resources` or store them in variables.

clonder committed:
Added library resourses and fixed configuration file

clonder commented:
Hi @alexandra-valeanu,

Thank you for reviewing this package. I’ve added resources to the dependencies—sorry for the inconvenience.

I was initially testing this within the same directory, so the `pyproject.toml` was always in the path. My plan was to include it in the build when publishing to PyPI in the next iteration.

We can use resources also, I’ve updated it to use the configuration files as per this commit.

Thanks again for your feedback!

alexandra-valeanu commented:
Hey @clonder! All good, it works now. Could you change the documentation in `docs/chapters/features/usage.rst`?

clonder committed:
Changed the documentation for kinex init

clonder commented:
Sure!

alexandra-valeanu commented on =0.12.:
```diff
@@ -0,0 +1,65 @@
+Collecting esbonio
+  Downloading esbonio-0.16.5-py3-none-any.whl.metadata (4.7 kB)
+Collecting sphinx (from esbonio)
+  Downloading sphinx-8.1.3-py3-none-any.whl.metadata (6.4 kB)
+Requirement already satisfied: platformdirs in /home/clony/v/lib/python3.12/site-packages (from esbonio) (4.3.6)
+Collecting pygls<2,>=1.1.0 (from esbonio)
+  Downloading pygls-1.3.1-py3-none-any.whl.metadata (4.7 kB)
```
@clonder was this file added by mistake?

clonder replied:
Yes

clonder committed:
Deleted file

Accio commented:
thank you both!
